### PR TITLE
Improve readability: add and use a script to add comments after `#else` and `#endif`

### PR DESCRIPTION
**Pull request by lyakh**
made with AI

### CODE REVIEW

**Review by ujfalusi — APPROVED** [review]

File: scripts/add-endif-comments.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+def compute_comment_state(lines):
+    """Return a list, parallel to lines, telling whether each line starts
+    inside an unterminated /* */ block comment. Used so that #if-like text
+    inside a comment is never mistaken for a real directive."""
+    states = []
+    in_block_comment = False
+    for line in lines:
+        states.append(in_block_comment)
+        i, length = 0, len(line)
+        in_string = in_char = False
```
[the remark this answers is not in the document]
Comment (lyakh) [reply scripts/add-endif-comments.py:53]: these seem very edge cases, and anyway after application manual review is recommended

**Comment by intel-sofci** [comment]
### PR 11191: test results

Run date: 2026-09-15 08:18 UTC

Tested commit: [d5b40286515b88c3bd3f8ed49b739a3e799c6f52](https://github.com/thesofproject/sof/commit/d5b40286515b88c3bd3f8ed49b739a3e799c6f52)

mtl pass rate lnl pass rate ptl pass rate wcl pass rate nvl pass rate

**Review by tmleman — APPROVED** [review]
